lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -838,11 +865,17 @@ sub fix_tests{
     my $table = $check->table;
     my $where = $check->id_where;
     my @edits = @{$check->columns};
+    # If we are inserting and id is displayed, we want to insert
+    # at this exact location
+    my $id_displayed = $check->{insert}
+                and grep( /^$check->{id_column}$/, @{$check->display_cols} );
 
     my $query;
-    if ($request->{insert}) {
-        my $columns = join(', ', map { $dbh->quote_identifier($_) } @edits);
-        my $values = join(', ', map { '?' } @edits);
+    if ($check->{insert}) {
+        my @id = $id_displayed ? $check->{id_column} : ();
```

Comment:
I'm sorry, I don't understand the bit which says `If we insert and do not display, it should be obvious that we do not have referrals at this moment.`: why does displaying have anything to do with having references or not? I mean, I can choose not to display the exact reference number, yet to allow data entry for the description of it, right?